src/main/java/net/lldp/checksims/token/TokenList.java

```diff
@@ -160,4 +166,14 @@ public String toString() {
     public int hashCode() {
         return type.hashCode() ^ super.hashCode();
     }
+
+    public int numValid() {
```

Comment:
This could be done as a one-liner with streams. Streams are awesome.

`return (int)this.stream().filter(Token::isValid).count();`